litestar/_asgi/asgi_router.py

```diff
@@ -83,9 +86,15 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
             path = path.split(root_path, maxsplit=1)[-1]
         normalized_path = normalize_path(path)
 
-        asgi_app, scope["route_handler"], scope["path"], scope["path_params"] = self.handle_routing(
-            path=normalized_path, method=scope.get("method")
-        )
+        try:
+            asgi_app, scope["route_handler"], scope["path"], scope["path_params"] = self.handle_routing(
+                path=normalized_path, method=scope.get("method")
+            )
+        except Exception:
+            ScopeState.from_scope(scope).exception_handlers = self._app_exception_handlers
+            raise
+        else:
+            ScopeState.from_scope(scope).exception_handlers = scope["route_handler"].resolve_exception_handlers()
```

Comment:
```suggestion
            asgi_app, route_handler, scope["path"], scope["path_params"] = self.handle_routing(
                path=normalized_path, method=scope.get("method")
            )
            scope["route_handler"] = route_handler
            ScopeState.from_scope(scope).exception_handlers = route_handler.resolve_exception_handlers()
        except Exception:
            ScopeState.from_scope(scope).exception_handlers = self._app_exception_handlers
            raise          
```

Avoiding one dictionary loopkup :P